Re-forked from scratch prior to pull request

carterscottm opened the pull request:
I cleaned up my commit list by starting over from scratch and bringing my files back into the new fork.

lethosor commented:
Are these the changes you want to merge in, then?

It's generally a good idea to keep images and other binary files out unless there's a good reason to have them in the repo (yeah, there is at least one already here), but these are small enough that I'm not super concerned.

PeridexisErrant commented:
This will just be an unorganised list of comments, but here goes:

- in PyLNP.json you switched spaces to tabs, which needlessly confuses the diff
- the LNP/Baselines dir will be created automatically by PyLNP; it shouldn't be tracked here
- the DFHack init extras should be rechecked with reference to DFHack/dfhack#970 - @lethosor might be merging many of them soon.  In general, I prefer to fix problems upstream wherever possible.
- PyLNP now handles `mouse.png` and `font.ttf`, so we don't need to track those either ([commit](https://bitbucket.org/Pidgeot/python-lnp/commits/42818ebb7fb56e10ad2f41c9c899640c699e016e))

And last but not least, thanks so much for taking the time to contribute :)

lethosor commented:
I'll comment on DFHack init stuff.

@PeridexisErrant - By the way, this is Enay from the forums, the maintainer of one of (?) the newer Linux packs. (He has a different name here, so I wasn't sure if you'd been following that thread and knew or not.)

lethosor commented on extras/dfhack_extras.init:
```diff
@@ -0,0 +1,21 @@
+
+# keybindings for Lethosor's scripts
+keybinding add Alt-S@title settings-manager
+keybinding add Alt-S@dwarfmode/Default settings-manager
```
These two bindings should be getting merged soon.

lethosor commented on extras/dfhack_extras.init:
```diff
@@ -0,0 +1,21 @@
+# keybindings for Lethosor's scripts
+keybinding add Alt-S@title settings-manager
+keybinding add Alt-S@dwarfmode/Default settings-manager
+keybinding add Alt-Q@jobmanagement manager-quantity
```
The `manager-quantity` script was replaced by `tweak manager-quantity`, which was then removed in 0.43.01 since there's no longer a limit of 30 in vanilla DF.

lethosor commented on extras/dfhack_extras.init:
```diff
@@ -0,0 +1,21 @@
+keybinding add Alt-W@overallstatus "gui/workflow status"
+
+# enable Lethosor's scripts
+embark-tools enable sand sticky
```
This one is in the default dfhack.init (except with `mouse` and without `sticky`)

lethosor commented on extras/dfhack_extras.init:
```diff
@@ -0,0 +1,21 @@
+multilevel 0
+
+# extended status screen (bedrooms page)
+enable gui/extended-status
```
Should be getting merged too.

PeridexisErrant commented:
@lethosor, @carterscottm - how about we just drop init changes from this pull, and I'll take responsibility for updating them after the DFHack pull?  I'll be doing that for myself anyway, and it would be a good step towards my using this base again.

carterscottm commented:
That sounds good to me.

As for the baselines issue you mentioned, I think there's an issue with PyLNP that causes the baselines to not be extracted from the .zip file the way it does in Windows.  I created the directory in my fork because I have my script download the baselines  and package them in with the archive I post on DFFD.  I probably should've submitted an issue on bitbucket about it in all honesty.

PeridexisErrant commented:
Baselines problem [reported upstream](https://bitbucket.org/Pidgeot/python-lnp/issues/124/).

I could keep the minor quibbles coming for a while, but given that they're minor I think it's fine.  What's the point of a shared core with *no users*?

(though next time you're editing a pull request, read up on `git rebase` with the `--squash` and/or `--interactive` options :wink:)

PeridexisErrant commented:
DFHack/dfhack#970 has been merged, largely obsoleting the init changes proposed here.

carterscottm commented:
I took a look at DFHack/dfhack#970 and removed those changes from my PR.  (Is this the right action to take, or can those changes be left out during the merge?)  I also removed some files from extras/hack/scripts that I neglected to pull out earlier.  They're just scripts that are already included in DFHack and don't need to be duplicated.

As for the manager-quantity script in extras/hack/scripts, I kept it in there because it allows you to change how many items are being produced in the jobs list, which is really handy once you have a job set up with a bunch of conditions and you realize you need to increase/decrease the amount being produced.

I went and rebased my local repo and squashed a bunch of my commits.  I think it's ready to push back up to carterscottm/LNP-shared-core:master.  Just checking, is that the next step in this process?  If I have to do `git push -f` to get it to rewrite history, is that going to screw anything up?  My apologies, I'm still very green with git.

lethosor commented:
Well, I completely missed that aspect of manager-quantity, so I'll re-add it to the dfhack/scripts repo (probably as gui/manager-quantity instead, though).

`git push -f` shouldn't break this PR, but I would compare what you have to 7a37c81 just to make sure you didn't accidentally leave anything out when you rebased your branch.

PeridexisErrant commented:
Fantastic work!
